Simplify library dependencies on `compiler-builtins`

tgross35 opened the pull request:
The three panic-related library crates need to have access to `core`, and `compiler-builtins` needs to be in the crate graph. Rather than specifying both dependencies, switch these crates to use `rustc-std-workspace-core` which already does this.

This means there is now a single place that the `compiler-builtins` dependency needs to get configured, for everything other than `alloc` and `std`.

The second commit removes `compiler-builtins` from `std` (more details in the message).

<!-- homu-ignore:start -->
r? @ghost
<!-- homu-ignore:end -->

tgross35 committed:
Use `core` via `rustc-std-workspace-core` in `library/panic*`

The three panic-related library crates need to have access to `core`,
and `compiler-builtins` needs to be in the crate graph. Rather than
specifying both dependencies, switch these crates to use
`rustc-std-workspace-core` which already does this.

This means there is now a single place that the `compiler-builtins`
dependency needs to get configured, for everything other than `alloc`
and `std`.

tgross35 committed:
Remove `std`'s dependency on `compiler-builtins`

`compiler-builtins` is already in the crate graph via `alloc`, and all
features related to `compiler-builtins` goes through `alloc`. There
isn't any reason that `std` needs this direct dependency, so remove it.

rust-log-analyzer commented:
The job **`aarch64-gnu-llvm-19-2`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/16625120261/job/47039487136) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/47039487136) [(plain)](https://github.com/rust-lang/rust/commit/38dea0f6242a6acf852bb6b5390685fe2c4030be/checks/47039487136/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
##[group]Building tool rustdoc_tool_binary (stage1 -> stage2, aarch64-unknown-linux-gnu)
##[endgroup]
##[group]Documenting stage1 library {alloc, compiler_builtins, core, panic_abort, panic_unwind, proc_macro, rustc-std-workspace-core, std, std_detect, sysroot, test, unwind} in HTML format (aarch64-unknown-linux-gnu)
##[endgroup]
Doc path: /tmp/.tmpA2XZUC/build/tmp-dry-run/aarch64-unknown-linux-gnu/doc/std/index.html
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ Snapshot Summary ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
Snapshot: alias_and_path_for_library-5
Source: src/bootstrap/src/core/builder/tests.rs:996
────────────────────────────────────────────────────────────────────────────────
Expression: ctx.config("doc").paths(&["library", "core"]).render_steps()
────────────────────────────────────────────────────────────────────────────────
-old snapshot
+new results
────────────┬───────────────────────────────────────────────────────────────────
    0     0 │ [build] llvm <host>
    1     1 │ [build] rustc 0 <host> -> rustc 1 <host>
    2     2 │ [build] rustdoc 0 <host>
    3       │-[doc] std 1 <host> crates=[alloc,compiler_builtins,core,panic_abort,panic_unwind,proc_macro,std,std_detect,sysroot,test,unwind]
          3 │+[doc] std 1 <host> crates=[alloc,compiler_builtins,core,panic_abort,panic_unwind,proc_macro,rustc-std-workspace-core,std,std_detect,sysroot,test,unwind]
────────────┴───────────────────────────────────────────────────────────────────
Stopped on the first failure. Run `cargo insta test` to run all snapshots.

thread 'core::builder::tests::snapshot::alias_and_path_for_library' panicked at /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/insta-1.43.1/src/runtime.rs:679:13:
snapshot assertion for 'alias_and_path_for_library-5' failed in line 996
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- core::builder::tests::snapshot::dist_all_cross stdout ----
Running test in /tmp/.tmpJ6KOob
Generating unstable book md files (i686-unknown-haiku)
##[group]Building stage1 tool unstable-book-gen (aarch64-unknown-linux-gnu)
##[endgroup]
##[group]Building stage1 tool rustbook (aarch64-unknown-linux-gnu)
##[endgroup]
Doc path: /tmp/.tmpJ6KOob/build/tmp-dry-run/i686-unknown-haiku/doc/unstable-book/index.html
##[group]Building LLVM for aarch64-unknown-linux-gnu
##[endgroup]
##[group]Building compiler artifacts (stage0 -> stage1, aarch64-unknown-linux-gnu)
##[endgroup]
Creating a sysroot for stage1 compiler (use `rustup toolchain link 'name' build/host/stage1`)
---
##[endgroup]
Creating a sysroot for stage2 compiler (use `rustup toolchain link 'name' build/host/stage2`)
##[group]Building stage2 tool wasm-component-ld (aarch64-unknown-linux-gnu)
##[endgroup]
Doc path: /tmp/.tmpJ6KOob/build/tmp-dry-run/i686-unknown-haiku/doc/book/index.html
##[group]Documenting stage2 book redirect pages (aarch64-unknown-linux-gnu -> i686-unknown-haiku)
##[endgroup]
##[group]Building tool rustdoc_tool_binary (stage2 -> stage3, aarch64-unknown-linux-gnu)
##[endgroup]
##[group]Documenting stage2 book redirect pages (aarch64-unknown-linux-gnu -> i686-unknown-haiku) (continued)
##[endgroup]
##[group]Documenting stage2 standalone (aarch64-unknown-linux-gnu -> i686-unknown-haiku)
Doc path: /tmp/.tmpJ6KOob/build/tmp-dry-run/i686-unknown-haiku/doc/index.html
##[endgroup]
##[group]Documenting stage2 library {alloc, compiler_builtins, core, panic_abort, panic_unwind, proc_macro, rustc-std-workspace-core, std, std_detect, sysroot, test, unwind} in HTML format (aarch64-unknown-linux-gnu -> i686-unknown-haiku)
##[endgroup]
Doc path: /tmp/.tmpJ6KOob/build/tmp-dry-run/i686-unknown-haiku/doc/alloc/index.html
Documenting error index (i686-unknown-haiku)
##[group]Building tool error_index_generator (stage2 -> stage3, aarch64-unknown-linux-gnu)
##[endgroup]
Doc path: /tmp/.tmpJ6KOob/build/tmp-dry-run/i686-unknown-haiku/doc/nomicon/index.html
Uplifting library (stage1 -> stage2)
Doc path: /tmp/.tmpJ6KOob/build/tmp-dry-run/i686-unknown-haiku/doc/reference/index.html
Doc path: /tmp/.tmpJ6KOob/build/tmp-dry-run/i686-unknown-haiku/doc/rustdoc/index.html
Doc path: /tmp/.tmpJ6KOob/build/tmp-dry-run/i686-unknown-haiku/doc/rust-by-example/index.html
Generating lint docs (i686-unknown-haiku)
##[group]Building stage1 library artifacts (aarch64-unknown-linux-gnu -> i686-unknown-haiku)
##[endgroup]
Uplifting library (stage1:aarch64-unknown-linux-gnu -> stage2:i686-unknown-haiku)
##[group]Building stage1 tool lint-docs (aarch64-unknown-linux-gnu)
##[endgroup]
##[group]Running stage2 lint-docs (aarch64-unknown-linux-gnu -> i686-unknown-haiku)
##[endgroup]
Doc path: /tmp/.tmpJ6KOob/build/tmp-dry-run/i686-unknown-haiku/doc/rustc/index.html
Doc path: /tmp/.tmpJ6KOob/build/tmp-dry-run/i686-unknown-haiku/doc/cargo/index.html
Doc path: /tmp/.tmpJ6KOob/build/tmp-dry-run/i686-unknown-haiku/doc/clippy/index.html
Doc path: /tmp/.tmpJ6KOob/build/tmp-dry-run/i686-unknown-haiku/doc/embedded-book/index.html
Doc path: /tmp/.tmpJ6KOob/build/tmp-dry-run/i686-unknown-haiku/doc/edition-guide/index.html
Doc path: /tmp/.tmpJ6KOob/build/tmp-dry-run/i686-unknown-haiku/doc/style-guide/index.html
##[group]Documenting stage2 releases (aarch64-unknown-linux-gnu -> i686-unknown-haiku)
##[endgroup]
##[group]Building stage1 tool rust-installer (aarch64-unknown-linux-gnu)
##[endgroup]
Dist rust-docs-nightly-i686-unknown-haiku
##[group]Documenting stage2 library {alloc, compiler_builtins, core, panic_abort, panic_unwind, proc_macro, rustc-std-workspace-core, std, std_detect, sysroot, test, unwind} in JSON format (aarch64-unknown-linux-gnu -> i686-unknown-haiku)
##[endgroup]
Dist rust-docs-json-nightly-i686-unknown-haiku
##[group]Building LLVM for i686-unknown-haiku
##[endgroup]
##[group]Building compiler artifacts (stage1:aarch64-unknown-linux-gnu -> stage2:i686-unknown-haiku)
##[endgroup]
Creating a sysroot for stage2 compiler (i686-unknown-haiku) (use `rustup toolchain link 'name' build/i686-unknown-haiku/stage2`)
##[group]Building stage2 tool wasm-component-ld (aarch64-unknown-linux-gnu -> i686-unknown-haiku)
##[endgroup]
##[group]Building tool rustdoc_tool_binary (stage2:aarch64-unknown-linux-gnu -> stage3:i686-unknown-haiku)
##[endgroup]
##[group]Building tool rust-analyzer-proc-macro-srv (stage2:aarch64-unknown-linux-gnu -> stage3:i686-unknown-haiku)
##[endgroup]
---
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ Snapshot Summary ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
Snapshot: dist_all_cross
Source: src/bootstrap/src/core/builder/tests.rs:1261
────────────────────────────────────────────────────────────────────────────────
Expression: ctx.config("dist").hosts(&[TEST_TRIPLE_1]).targets(&[TEST_TRIPLE_1]).args(&["--set",
"rust.channel=nightly", "--set", "build.extended=true"]).render_steps()
────────────────────────────────────────────────────────────────────────────────
-old snapshot
+new results
────────────┬───────────────────────────────────────────────────────────────────
    5     5 │ [build] rustc 1 <host> -> std 1 <host>
    6     6 │ [build] rustc 1 <host> -> rustc 2 <host>
    7     7 │ [build] rustc 1 <host> -> WasmComponentLd 2 <host>
    8     8 │ [build] rustdoc 1 <host>
    9       │-[doc] std 2 <target1> crates=[alloc,compiler_builtins,core,panic_abort,panic_unwind,proc_macro,std,std_detect,sysroot,test,unwind]
          9 │+[doc] std 2 <target1> crates=[alloc,compiler_builtins,core,panic_abort,panic_unwind,proc_macro,rustc-std-workspace-core,std,std_detect,sysroot,test,unwind]
   10    10 │ [build] rustc 2 <host> -> std 2 <host>
   11    11 │ [build] rustc 1 <host> -> std 1 <target1>
   12    12 │ [build] rustc 2 <host> -> std 2 <target1>
   13    13 │ [build] rustc 0 <host> -> LintDocs 1 <host>
────────────┴───────────────────────────────────────────────────────────────────
Stopped on the first failure. Run `cargo insta test` to run all snapshots.

thread 'core::builder::tests::snapshot::dist_all_cross' panicked at /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/insta-1.43.1/src/runtime.rs:679:13:
snapshot assertion for 'dist_all_cross' failed in line 1261
---
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ Snapshot Summary ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
Snapshot: dist_baseline
Source: src/bootstrap/src/core/builder/tests.rs:1040
────────────────────────────────────────────────────────────────────────────────
Expression: ctx.config("dist").render_steps()
────────────────────────────────────────────────────────────────────────────────
-old snapshot
+new results
────────────┬───────────────────────────────────────────────────────────────────
    3     3 │ [build] rustc 0 <host> -> rustc 1 <host>
    4     4 │ [build] rustc 1 <host> -> std 1 <host>
    5     5 │ [build] rustc 1 <host> -> rustc 2 <host>
    6     6 │ [build] rustdoc 1 <host>
    7       │-[doc] std 2 <host> crates=[alloc,compiler_builtins,core,panic_abort,panic_unwind,proc_macro,std,std_detect,sysroot,test,unwind]
          7 │+[doc] std 2 <host> crates=[alloc,compiler_builtins,core,panic_abort,panic_unwind,proc_macro,rustc-std-workspace-core,std,std_detect,sysroot,test,unwind]
    8     8 │ [build] rustc 2 <host> -> std 2 <host>
    9     9 │ [build] rustc 0 <host> -> LintDocs 1 <host>
   10    10 │ [build] rustc 0 <host> -> RustInstaller 1 <host>
   11    11 │ [dist] docs <host>
────────────┴───────────────────────────────────────────────────────────────────
Stopped on the first failure. Run `cargo insta test` to run all snapshots.

thread 'core::builder::tests::snapshot::dist_baseline' panicked at /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/insta-1.43.1/src/runtime.rs:679:13:
snapshot assertion for 'dist_baseline' failed in line 1040
---
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ Snapshot Summary ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
Snapshot: dist_extended
Source: src/bootstrap/src/core/builder/tests.rs:1069
────────────────────────────────────────────────────────────────────────────────
Expression: ctx.config("dist").args(&["--set", "build.extended=true", "--set",
"rust.llvm-bitcode-linker=true", "--set", "rust.lld=true",]).render_steps()
────────────────────────────────────────────────────────────────────────────────
-old snapshot
+new results
────────────┬───────────────────────────────────────────────────────────────────
    9     9 │ [build] rustc 1 <host> -> LldWrapper 2 <host>
   10    10 │ [build] rustc 1 <host> -> WasmComponentLd 2 <host>
   11    11 │ [build] rustc 1 <host> -> LlvmBitcodeLinker 2 <host>
   12    12 │ [build] rustdoc 1 <host>
   13       │-[doc] std 2 <host> crates=[alloc,compiler_builtins,core,panic_abort,panic_unwind,proc_macro,std,std_detect,sysroot,test,unwind]
         13 │+[doc] std 2 <host> crates=[alloc,compiler_builtins,core,panic_abort,panic_unwind,proc_macro,rustc-std-workspace-core,std,std_detect,sysroot,test,unwind]
   14    14 │ [build] rustc 2 <host> -> std 2 <host>
   15    15 │ [build] rustc 0 <host> -> LintDocs 1 <host>
   16    16 │ [build] rustc 0 <host> -> RustInstaller 1 <host>
   17    17 │ [dist] docs <host>
────────────┴───────────────────────────────────────────────────────────────────
Stopped on the first failure. Run `cargo insta test` to run all snapshots.

thread 'core::builder::tests::snapshot::dist_extended' panicked at /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/insta-1.43.1/src/runtime.rs:679:13:
snapshot assertion for 'dist_extended' failed in line 1069
---
##[group]Building stage1 tool unstable-book-gen (aarch64-unknown-linux-gnu)
##[endgroup]
##[group]Building stage1 tool rustbook (aarch64-unknown-linux-gnu)
##[endgroup]
Doc path: /tmp/.tmp7gvdhH/build/tmp-dry-run/i686-unknown-haiku/doc/unstable-book/index.html
##[group]Building LLVM for aarch64-unknown-linux-gnu
##[endgroup]
##[group]Building compiler artifacts (stage0 -> stage1, aarch64-unknown-linux-gnu)
##[endgroup]
Creating a sysroot for stage1 compiler (use `rustup toolchain link 'name' build/host/stage1`)
##[group]Building stage1 library artifacts (aarch64-unknown-linux-gnu)
##[endgroup]
##[group]Building compiler artifacts (stage1 -> stage2, aarch64-unknown-linux-gnu)
##[endgroup]
Creating a sysroot for stage2 compiler (use `rustup toolchain link 'name' build/host/stage2`)
Doc path: /tmp/.tmp7gvdhH/build/tmp-dry-run/i686-unknown-haiku/doc/book/index.html
##[group]Documenting stage2 book redirect pages (aarch64-unknown-linux-gnu -> i686-unknown-haiku)
##[endgroup]
##[group]Building tool rustdoc_tool_binary (stage2 -> stage3, aarch64-unknown-linux-gnu)
##[endgroup]
##[group]Documenting stage2 book redirect pages (aarch64-unknown-linux-gnu -> i686-unknown-haiku) (continued)
##[endgroup]
##[group]Documenting stage2 standalone (aarch64-unknown-linux-gnu -> i686-unknown-haiku)
Doc path: /tmp/.tmp7gvdhH/build/tmp-dry-run/i686-unknown-haiku/doc/index.html
##[endgroup]
##[group]Documenting stage2 library {alloc, compiler_builtins, core, panic_abort, panic_unwind, proc_macro, rustc-std-workspace-core, std, std_detect, sysroot, test, unwind} in HTML format (aarch64-unknown-linux-gnu -> i686-unknown-haiku)
##[endgroup]
Doc path: /tmp/.tmp7gvdhH/build/tmp-dry-run/i686-unknown-haiku/doc/alloc/index.html
Doc path: /tmp/.tmp7gvdhH/build/tmp-dry-run/i686-unknown-haiku/doc/nomicon/index.html
Uplifting library (stage1 -> stage2)
Doc path: /tmp/.tmp7gvdhH/build/tmp-dry-run/i686-unknown-haiku/doc/reference/index.html
Doc path: /tmp/.tmp7gvdhH/build/tmp-dry-run/i686-unknown-haiku/doc/rustdoc/index.html
Doc path: /tmp/.tmp7gvdhH/build/tmp-dry-run/i686-unknown-haiku/doc/rust-by-example/index.html
Doc path: /tmp/.tmp7gvdhH/build/tmp-dry-run/i686-unknown-haiku/doc/cargo/index.html
Doc path: /tmp/.tmp7gvdhH/build/tmp-dry-run/i686-unknown-haiku/doc/clippy/index.html
Doc path: /tmp/.tmp7gvdhH/build/tmp-dry-run/i686-unknown-haiku/doc/embedded-book/index.html
Doc path: /tmp/.tmp7gvdhH/build/tmp-dry-run/i686-unknown-haiku/doc/edition-guide/index.html
Doc path: /tmp/.tmp7gvdhH/build/tmp-dry-run/i686-unknown-haiku/doc/style-guide/index.html
##[group]Documenting stage2 releases (aarch64-unknown-linux-gnu -> i686-unknown-haiku)
##[endgroup]
##[group]Building stage1 tool rust-installer (aarch64-unknown-linux-gnu)
##[endgroup]
Dist rust-docs-1.90.0-dev-i686-unknown-haiku
##[group]Documenting stage2 library {alloc, compiler_builtins, core, panic_abort, panic_unwind, proc_macro, rustc-std-workspace-core, std, std_detect, sysroot, test, unwind} in JSON format (aarch64-unknown-linux-gnu -> i686-unknown-haiku)
##[endgroup]
Dist rust-docs-json-1.90.0-dev-i686-unknown-haiku
##[group]Building stage2 library artifacts (aarch64-unknown-linux-gnu -> i686-unknown-haiku)
##[endgroup]
Dist rust-std-1.90.0-dev-i686-unknown-haiku
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ Snapshot Summary ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
Snapshot: dist_with_empty_host
Source: src/bootstrap/src/core/builder/tests.rs:1232
────────────────────────────────────────────────────────────────────────────────
Expression: ctx.config("dist").hosts(&[]).targets(&[TEST_TRIPLE_1]).render_steps()
────────────────────────────────────────────────────────────────────────────────
-old snapshot
+new results
────────────┬───────────────────────────────────────────────────────────────────
    3     3 │ [build] rustc 0 <host> -> rustc 1 <host>
    4     4 │ [build] rustc 1 <host> -> std 1 <host>
    5     5 │ [build] rustc 1 <host> -> rustc 2 <host>
    6     6 │ [build] rustdoc 1 <host>
    7       │-[doc] std 2 <target1> crates=[alloc,compiler_builtins,core,panic_abort,panic_unwind,proc_macro,std,std_detect,sysroot,test,unwind]
          7 │+[doc] std 2 <target1> crates=[alloc,compiler_builtins,core,panic_abort,panic_unwind,proc_macro,rustc-std-workspace-core,std,std_detect,sysroot,test,unwind]
    8     8 │ [build] rustc 2 <host> -> std 2 <host>
    9     9 │ [build] rustc 0 <host> -> RustInstaller 1 <host>
   10    10 │ [dist] docs <target1>
   11    11 │ [doc] std 2 <target1> crates=[]
────────────┴───────────────────────────────────────────────────────────────────
Stopped on the first failure. Run `cargo insta test` to run all snapshots.

thread 'core::builder::tests::snapshot::dist_with_empty_host' panicked at /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/insta-1.43.1/src/runtime.rs:679:13:
snapshot assertion for 'dist_with_empty_host' failed in line 1232
---
##[group]Building stage1 tool unstable-book-gen (aarch64-unknown-linux-gnu)
##[endgroup]
##[group]Building stage1 tool rustbook (aarch64-unknown-linux-gnu)
##[endgroup]
Doc path: /tmp/.tmpRxSFK6/build/tmp-dry-run/aarch64-unknown-linux-gnu/doc/unstable-book/index.html
Generating unstable book md files (i686-unknown-haiku)
##[group]Building LLVM for aarch64-unknown-linux-gnu
##[endgroup]
##[group]Building compiler artifacts (stage0 -> stage1, aarch64-unknown-linux-gnu)
##[endgroup]
Creating a sysroot for stage1 compiler (use `rustup toolchain link 'name' build/host/stage1`)
##[group]Building stage1 library artifacts (aarch64-unknown-linux-gnu)
##[endgroup]
##[group]Building compiler artifacts (stage1 -> stage2, aarch64-unknown-linux-gnu)
##[endgroup]
Creating a sysroot for stage2 compiler (use `rustup toolchain link 'name' build/host/stage2`)
Doc path: /tmp/.tmpRxSFK6/build/tmp-dry-run/aarch64-unknown-linux-gnu/doc/book/index.html
##[group]Documenting stage2 book redirect pages (aarch64-unknown-linux-gnu)
##[endgroup]
##[group]Building tool rustdoc_tool_binary (stage2 -> stage3, aarch64-unknown-linux-gnu)
##[endgroup]
##[group]Documenting stage2 book redirect pages (aarch64-unknown-linux-gnu) (continued)
##[endgroup]
##[group]Documenting stage2 standalone (aarch64-unknown-linux-gnu)
Doc path: /tmp/.tmpRxSFK6/build/tmp-dry-run/aarch64-unknown-linux-gnu/doc/index.html
##[endgroup]
##[group]Documenting stage2 library {alloc, compiler_builtins, core, panic_abort, panic_unwind, proc_macro, rustc-std-workspace-core, std, std_detect, sysroot, test, unwind} in HTML format (aarch64-unknown-linux-gnu)
##[endgroup]
Doc path: /tmp/.tmpRxSFK6/build/tmp-dry-run/aarch64-unknown-linux-gnu/doc/alloc/index.html
Documenting error index (aarch64-unknown-linux-gnu)
##[group]Building tool error_index_generator (stage2 -> stage3, aarch64-unknown-linux-gnu)
##[endgroup]
Documenting error index (i686-unknown-haiku)
Doc path: /tmp/.tmpRxSFK6/build/tmp-dry-run/aarch64-unknown-linux-gnu/doc/nomicon/index.html
Uplifting library (stage1 -> stage2)
Doc path: /tmp/.tmpRxSFK6/build/tmp-dry-run/aarch64-unknown-linux-gnu/doc/reference/index.html
Doc path: /tmp/.tmpRxSFK6/build/tmp-dry-run/aarch64-unknown-linux-gnu/doc/rustdoc/index.html
Doc path: /tmp/.tmpRxSFK6/build/tmp-dry-run/aarch64-unknown-linux-gnu/doc/rust-by-example/index.html
Generating lint docs (aarch64-unknown-linux-gnu)
##[group]Building stage1 tool lint-docs (aarch64-unknown-linux-gnu)
##[endgroup]
##[group]Running stage2 lint-docs (aarch64-unknown-linux-gnu)
##[endgroup]
Doc path: /tmp/.tmpRxSFK6/build/tmp-dry-run/aarch64-unknown-linux-gnu/doc/rustc/index.html
Generating lint docs (i686-unknown-haiku)
##[group]Building stage1 library artifacts (aarch64-unknown-linux-gnu -> i686-unknown-haiku)
##[endgroup]
Uplifting library (stage1:aarch64-unknown-linux-gnu -> stage2:i686-unknown-haiku)
##[group]Running stage2 lint-docs (aarch64-unknown-linux-gnu -> i686-unknown-haiku)
##[endgroup]
Doc path: /tmp/.tmpRxSFK6/build/tmp-dry-run/i686-unknown-haiku/doc/rustc/index.html
Doc path: /tmp/.tmpRxSFK6/build/tmp-dry-run/aarch64-unknown-linux-gnu/doc/cargo/index.html
Doc path: /tmp/.tmpRxSFK6/build/tmp-dry-run/aarch64-unknown-linux-gnu/doc/clippy/index.html
Doc path: /tmp/.tmpRxSFK6/build/tmp-dry-run/aarch64-unknown-linux-gnu/doc/embedded-book/index.html
Doc path: /tmp/.tmpRxSFK6/build/tmp-dry-run/aarch64-unknown-linux-gnu/doc/edition-guide/index.html
Doc path: /tmp/.tmpRxSFK6/build/tmp-dry-run/aarch64-unknown-linux-gnu/doc/style-guide/index.html
##[group]Documenting stage2 releases (aarch64-unknown-linux-gnu)
##[endgroup]
##[group]Building stage1 tool rust-installer (aarch64-unknown-linux-gnu)
##[endgroup]
Dist rust-docs-1.90.0-dev-aarch64-unknown-linux-gnu
---
##[group]Building LLVM for i686-unknown-haiku
##[endgroup]
##[group]Building compiler artifacts (stage1:aarch64-unknown-linux-gnu -> stage2:i686-unknown-haiku)
##[endgroup]
Creating a sysroot for stage2 compiler (i686-unknown-haiku) (use `rustup toolchain link 'name' build/i686-unknown-haiku/stage2`)
##[group]Building tool rustdoc_tool_binary (stage2:aarch64-unknown-linux-gnu -> stage3:i686-unknown-haiku)
##[endgroup]
Dist rustc-1.90.0-dev-i686-unknown-haiku
Dist rust-std-1.90.0-dev-aarch64-unknown-linux-gnu
Dist rustc-dev-1.90.0-dev-aarch64-unknown-linux-gnu
Dist rustc-dev-1.90.0-dev-i686-unknown-haiku
Dist rust-src-1.90.0-dev
Dist rust-dev-1.90.0-dev-aarch64-unknown-linux-gnu
Dist rust-dev-1.90.0-dev-i686-unknown-haiku
Dist rustc-1.90.0-dev-src
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ Snapshot Summary ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
Snapshot: dist_with_hosts
Source: src/bootstrap/src/core/builder/tests.rs:1153
────────────────────────────────────────────────────────────────────────────────
Expression: ctx.config("dist").hosts(&[&host_target(),
TEST_TRIPLE_1]).targets(&[&host_target()]).render_steps()
────────────────────────────────────────────────────────────────────────────────
-old snapshot
+new results
────────────┬───────────────────────────────────────────────────────────────────
    3     3 │ [build] rustc 0 <host> -> rustc 1 <host>
    4     4 │ [build] rustc 1 <host> -> std 1 <host>
    5     5 │ [build] rustc 1 <host> -> rustc 2 <host>
    6     6 │ [build] rustdoc 1 <host>
    7       │-[doc] std 2 <host> crates=[alloc,compiler_builtins,core,panic_abort,panic_unwind,proc_macro,std,std_detect,sysroot,test,unwind]
          7 │+[doc] std 2 <host> crates=[alloc,compiler_builtins,core,panic_abort,panic_unwind,proc_macro,rustc-std-workspace-core,std,std_detect,sysroot,test,unwind]
    8     8 │ [build] rustc 2 <host> -> std 2 <host>
    9     9 │ [build] rustc 0 <host> -> LintDocs 1 <host>
   10    10 │ [build] rustc 1 <host> -> std 1 <target1>
   11    11 │ [build] rustc 2 <host> -> std 2 <target1>
────────────┴───────────────────────────────────────────────────────────────────
Stopped on the first failure. Run `cargo insta test` to run all snapshots.

thread 'core::builder::tests::snapshot::dist_with_hosts' panicked at /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/insta-1.43.1/src/runtime.rs:679:13:
snapshot assertion for 'dist_with_hosts' failed in line 1153
---
##[group]Building stage1 tool unstable-book-gen (aarch64-unknown-linux-gnu)
##[endgroup]
##[group]Building stage1 tool rustbook (aarch64-unknown-linux-gnu)
##[endgroup]
Doc path: /tmp/.tmp17DbE2/build/tmp-dry-run/aarch64-unknown-linux-gnu/doc/unstable-book/index.html
Generating unstable book md files (i686-unknown-haiku)
Doc path: /tmp/.tmp17DbE2/build/tmp-dry-run/i686-unknown-haiku/doc/unstable-book/index.html
##[group]Building LLVM for aarch64-unknown-linux-gnu
##[endgroup]
##[group]Building compiler artifacts (stage0 -> stage1, aarch64-unknown-linux-gnu)
##[endgroup]
Creating a sysroot for stage1 compiler (use `rustup toolchain link 'name' build/host/stage1`)
---
##[group]Building tool rustdoc_tool_binary (stage2 -> stage3, aarch64-unknown-linux-gnu)
##[endgroup]
##[group]Documenting stage2 book redirect pages (aarch64-unknown-linux-gnu) (continued)
##[endgroup]
Doc path: /tmp/.tmp17DbE2/build/tmp-dry-run/i686-unknown-haiku/doc/book/index.html
##[group]Documenting stage2 book redirect pages (aarch64-unknown-linux-gnu -> i686-unknown-haiku)
##[endgroup]
##[group]Documenting stage2 standalone (aarch64-unknown-linux-gnu)
Doc path: /tmp/.tmp17DbE2/build/tmp-dry-run/aarch64-unknown-linux-gnu/doc/index.html
##[endgroup]
##[group]Documenting stage2 standalone (aarch64-unknown-linux-gnu -> i686-unknown-haiku)
Doc path: /tmp/.tmp17DbE2/build/tmp-dry-run/i686-unknown-haiku/doc/index.html
##[endgroup]
##[group]Documenting stage2 library {alloc, compiler_builtins, core, panic_abort, panic_unwind, proc_macro, rustc-std-workspace-core, std, std_detect, sysroot, test, unwind} in HTML format (aarch64-unknown-linux-gnu)
##[endgroup]
Doc path: /tmp/.tmp17DbE2/build/tmp-dry-run/aarch64-unknown-linux-gnu/doc/alloc/index.html
##[group]Documenting stage2 library {alloc, compiler_builtins, core, panic_abort, panic_unwind, proc_macro, rustc-std-workspace-core, std, std_detect, sysroot, test, unwind} in HTML format (aarch64-unknown-linux-gnu -> i686-unknown-haiku)
##[endgroup]
Doc path: /tmp/.tmp17DbE2/build/tmp-dry-run/i686-unknown-haiku/doc/alloc/index.html
Documenting error index (aarch64-unknown-linux-gnu)
##[group]Building tool error_index_generator (stage2 -> stage3, aarch64-unknown-linux-gnu)
##[endgroup]
Doc path: /tmp/.tmp17DbE2/build/tmp-dry-run/aarch64-unknown-linux-gnu/doc/nomicon/index.html
Doc path: /tmp/.tmp17DbE2/build/tmp-dry-run/i686-unknown-haiku/doc/nomicon/index.html
Uplifting library (stage1 -> stage2)
Doc path: /tmp/.tmp17DbE2/build/tmp-dry-run/aarch64-unknown-linux-gnu/doc/reference/index.html
Doc path: /tmp/.tmp17DbE2/build/tmp-dry-run/i686-unknown-haiku/doc/reference/index.html
Doc path: /tmp/.tmp17DbE2/build/tmp-dry-run/aarch64-unknown-linux-gnu/doc/rustdoc/index.html
Doc path: /tmp/.tmp17DbE2/build/tmp-dry-run/i686-unknown-haiku/doc/rustdoc/index.html
Doc path: /tmp/.tmp17DbE2/build/tmp-dry-run/aarch64-unknown-linux-gnu/doc/rust-by-example/index.html
Doc path: /tmp/.tmp17DbE2/build/tmp-dry-run/i686-unknown-haiku/doc/rust-by-example/index.html
Generating lint docs (aarch64-unknown-linux-gnu)
##[group]Building stage1 tool lint-docs (aarch64-unknown-linux-gnu)
##[endgroup]
##[group]Running stage2 lint-docs (aarch64-unknown-linux-gnu)
##[endgroup]
Doc path: /tmp/.tmp17DbE2/build/tmp-dry-run/aarch64-unknown-linux-gnu/doc/rustc/index.html
Doc path: /tmp/.tmp17DbE2/build/tmp-dry-run/aarch64-unknown-linux-gnu/doc/cargo/index.html
Doc path: /tmp/.tmp17DbE2/build/tmp-dry-run/i686-unknown-haiku/doc/cargo/index.html
Doc path: /tmp/.tmp17DbE2/build/tmp-dry-run/aarch64-unknown-linux-gnu/doc/clippy/index.html
Doc path: /tmp/.tmp17DbE2/build/tmp-dry-run/i686-unknown-haiku/doc/clippy/index.html
Doc path: /tmp/.tmp17DbE2/build/tmp-dry-run/aarch64-unknown-linux-gnu/doc/embedded-book/index.html
Doc path: /tmp/.tmp17DbE2/build/tmp-dry-run/i686-unknown-haiku/doc/embedded-book/index.html
Doc path: /tmp/.tmp17DbE2/build/tmp-dry-run/aarch64-unknown-linux-gnu/doc/edition-guide/index.html
Doc path: /tmp/.tmp17DbE2/build/tmp-dry-run/i686-unknown-haiku/doc/edition-guide/index.html
Doc path: /tmp/.tmp17DbE2/build/tmp-dry-run/aarch64-unknown-linux-gnu/doc/style-guide/index.html
Doc path: /tmp/.tmp17DbE2/build/tmp-dry-run/i686-unknown-haiku/doc/style-guide/index.html
##[group]Documenting stage2 releases (aarch64-unknown-linux-gnu)
##[endgroup]
##[group]Documenting stage2 releases (aarch64-unknown-linux-gnu -> i686-unknown-haiku)
##[endgroup]
##[group]Building stage1 tool rust-installer (aarch64-unknown-linux-gnu)
##[endgroup]
Dist rust-docs-1.90.0-dev-aarch64-unknown-linux-gnu
Dist rust-docs-1.90.0-dev-i686-unknown-haiku
##[group]Documenting stage2 library {alloc, compiler_builtins, core, panic_abort, panic_unwind, proc_macro, rustc-std-workspace-core, std, std_detect, sysroot, test, unwind} in JSON format (aarch64-unknown-linux-gnu)
##[endgroup]
Dist rust-docs-json-1.90.0-dev-aarch64-unknown-linux-gnu
##[group]Documenting stage2 library {alloc, compiler_builtins, core, panic_abort, panic_unwind, proc_macro, rustc-std-workspace-core, std, std_detect, sysroot, test, unwind} in JSON format (aarch64-unknown-linux-gnu -> i686-unknown-haiku)
##[endgroup]
Dist rust-docs-json-1.90.0-dev-i686-unknown-haiku
Vendoring sources to "/checkout"
##[group]Building stage1 tool generate-copyright (aarch64-unknown-linux-gnu)
##[endgroup]
Dist rustc-1.90.0-dev-aarch64-unknown-linux-gnu
Dist rust-std-1.90.0-dev-aarch64-unknown-linux-gnu
##[group]Building stage2 library artifacts (aarch64-unknown-linux-gnu -> i686-unknown-haiku)
##[endgroup]
Dist rust-std-1.90.0-dev-i686-unknown-haiku
Dist rustc-dev-1.90.0-dev-aarch64-unknown-linux-gnu
Dist rust-src-1.90.0-dev
Dist rust-dev-1.90.0-dev-aarch64-unknown-linux-gnu
Dist rustc-1.90.0-dev-src
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ Snapshot Summary ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
Snapshot: dist_with_targets
Source: src/bootstrap/src/core/builder/tests.rs:1116
────────────────────────────────────────────────────────────────────────────────
Expression: ctx.config("dist").hosts(&[&host_target()]).targets(&[&host_target(),
TEST_TRIPLE_1]).render_steps()
────────────────────────────────────────────────────────────────────────────────
-old snapshot
+new results
────────────┬───────────────────────────────────────────────────────────────────
    3     3 │ [build] rustc 0 <host> -> rustc 1 <host>
    4     4 │ [build] rustc 1 <host> -> std 1 <host>
    5     5 │ [build] rustc 1 <host> -> rustc 2 <host>
    6     6 │ [build] rustdoc 1 <host>
    7       │-[doc] std 2 <host> crates=[alloc,compiler_builtins,core,panic_abort,panic_unwind,proc_macro,std,std_detect,sysroot,test,unwind]
    8       │-[doc] std 2 <target1> crates=[alloc,compiler_builtins,core,panic_abort,panic_unwind,proc_macro,std,std_detect,sysroot,test,unwind]
          7 │+[doc] std 2 <host> crates=[alloc,compiler_builtins,core,panic_abort,panic_unwind,proc_macro,rustc-std-workspace-core,std,std_detect,sysroot,test,unwind]
          8 │+[doc] std 2 <target1> crates=[alloc,compiler_builtins,core,panic_abort,panic_unwind,proc_macro,rustc-std-workspace-core,std,std_detect,sysroot,test,unwind]
    9     9 │ [build] rustc 2 <host> -> std 2 <host>
   10    10 │ [build] rustc 0 <host> -> LintDocs 1 <host>
   11    11 │ [build] rustc 0 <host> -> RustInstaller 1 <host>
   12    12 │ [dist] docs <host>
────────────┴───────────────────────────────────────────────────────────────────
Stopped on the first failure. Run `cargo insta test` to run all snapshots.

thread 'core::builder::tests::snapshot::dist_with_targets' panicked at /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/insta-1.43.1/src/runtime.rs:679:13:
snapshot assertion for 'dist_with_targets' failed in line 1116
---
##[group]Building stage1 tool unstable-book-gen (aarch64-unknown-linux-gnu)
##[endgroup]
##[group]Building stage1 tool rustbook (aarch64-unknown-linux-gnu)
##[endgroup]
Doc path: /tmp/.tmpZVqfNi/build/tmp-dry-run/aarch64-unknown-linux-gnu/doc/unstable-book/index.html
Generating unstable book md files (i686-unknown-haiku)
Doc path: /tmp/.tmpZVqfNi/build/tmp-dry-run/i686-unknown-haiku/doc/unstable-book/index.html
##[group]Building LLVM for aarch64-unknown-linux-gnu
##[endgroup]
##[group]Building compiler artifacts (stage0 -> stage1, aarch64-unknown-linux-gnu)
##[endgroup]
Creating a sysroot for stage1 compiler (use `rustup toolchain link 'name' build/host/stage1`)
---
##[group]Building tool rustdoc_tool_binary (stage2 -> stage3, aarch64-unknown-linux-gnu)
##[endgroup]
##[group]Documenting stage2 book redirect pages (aarch64-unknown-linux-gnu) (continued)
##[endgroup]
Doc path: /tmp/.tmpZVqfNi/build/tmp-dry-run/i686-unknown-haiku/doc/book/index.html
##[group]Documenting stage2 book redirect pages (aarch64-unknown-linux-gnu -> i686-unknown-haiku)
##[endgroup]
##[group]Documenting stage2 standalone (aarch64-unknown-linux-gnu)
Doc path: /tmp/.tmpZVqfNi/build/tmp-dry-run/aarch64-unknown-linux-gnu/doc/index.html
##[endgroup]
##[group]Documenting stage2 standalone (aarch64-unknown-linux-gnu -> i686-unknown-haiku)
Doc path: /tmp/.tmpZVqfNi/build/tmp-dry-run/i686-unknown-haiku/doc/index.html
##[endgroup]
##[group]Documenting stage2 library {alloc, compiler_builtins, core, panic_abort, panic_unwind, proc_macro, rustc-std-workspace-core, std, std_detect, sysroot, test, unwind} in HTML format (aarch64-unknown-linux-gnu)
##[endgroup]
Doc path: /tmp/.tmpZVqfNi/build/tmp-dry-run/aarch64-unknown-linux-gnu/doc/alloc/index.html
##[group]Documenting stage2 library {alloc, compiler_builtins, core, panic_abort, panic_unwind, proc_macro, rustc-std-workspace-core, std, std_detect, sysroot, test, unwind} in HTML format (aarch64-unknown-linux-gnu -> i686-unknown-haiku)
##[endgroup]
Doc path: /tmp/.tmpZVqfNi/build/tmp-dry-run/i686-unknown-haiku/doc/alloc/index.html
Documenting error index (aarch64-unknown-linux-gnu)
##[group]Building tool error_index_generator (stage2 -> stage3, aarch64-unknown-linux-gnu)
##[endgroup]
Documenting error index (i686-unknown-haiku)
Doc path: /tmp/.tmpZVqfNi/build/tmp-dry-run/aarch64-unknown-linux-gnu/doc/nomicon/index.html
Doc path: /tmp/.tmpZVqfNi/build/tmp-dry-run/i686-unknown-haiku/doc/nomicon/index.html
Uplifting library (stage1 -> stage2)
Doc path: /tmp/.tmpZVqfNi/build/tmp-dry-run/aarch64-unknown-linux-gnu/doc/reference/index.html
Doc path: /tmp/.tmpZVqfNi/build/tmp-dry-run/i686-unknown-haiku/doc/reference/index.html
Doc path: /tmp/.tmpZVqfNi/build/tmp-dry-run/aarch64-unknown-linux-gnu/doc/rustdoc/index.html
Doc path: /tmp/.tmpZVqfNi/build/tmp-dry-run/i686-unknown-haiku/doc/rustdoc/index.html
Doc path: /tmp/.tmpZVqfNi/build/tmp-dry-run/aarch64-unknown-linux-gnu/doc/rust-by-example/index.html
Doc path: /tmp/.tmpZVqfNi/build/tmp-dry-run/i686-unknown-haiku/doc/rust-by-example/index.html
Generating lint docs (aarch64-unknown-linux-gnu)
##[group]Building stage1 tool lint-docs (aarch64-unknown-linux-gnu)
##[endgroup]
##[group]Running stage2 lint-docs (aarch64-unknown-linux-gnu)
##[endgroup]
Doc path: /tmp/.tmpZVqfNi/build/tmp-dry-run/aarch64-unknown-linux-gnu/doc/rustc/index.html
Generating lint docs (i686-unknown-haiku)
##[group]Building stage1 library artifacts (aarch64-unknown-linux-gnu -> i686-unknown-haiku)
##[endgroup]
Uplifting library (stage1:aarch64-unknown-linux-gnu -> stage2:i686-unknown-haiku)
##[group]Running stage2 lint-docs (aarch64-unknown-linux-gnu -> i686-unknown-haiku)
##[endgroup]
Doc path: /tmp/.tmpZVqfNi/build/tmp-dry-run/i686-unknown-haiku/doc/rustc/index.html
Doc path: /tmp/.tmpZVqfNi/build/tmp-dry-run/aarch64-unknown-linux-gnu/doc/cargo/index.html
Doc path: /tmp/.tmpZVqfNi/build/tmp-dry-run/i686-unknown-haiku/doc/cargo/index.html
Doc path: /tmp/.tmpZVqfNi/build/tmp-dry-run/aarch64-unknown-linux-gnu/doc/clippy/index.html
Doc path: /tmp/.tmpZVqfNi/build/tmp-dry-run/i686-unknown-haiku/doc/clippy/index.html
Doc path: /tmp/.tmpZVqfNi/build/tmp-dry-run/aarch64-unknown-linux-gnu/doc/embedded-book/index.html
Doc path: /tmp/.tmpZVqfNi/build/tmp-dry-run/i686-unknown-haiku/doc/embedded-book/index.html
Doc path: /tmp/.tmpZVqfNi/build/tmp-dry-run/aarch64-unknown-linux-gnu/doc/edition-guide/index.html
Doc path: /tmp/.tmpZVqfNi/build/tmp-dry-run/i686-unknown-haiku/doc/edition-guide/index.html
Doc path: /tmp/.tmpZVqfNi/build/tmp-dry-run/aarch64-unknown-linux-gnu/doc/style-guide/index.html
Doc path: /tmp/.tmpZVqfNi/build/tmp-dry-run/i686-unknown-haiku/doc/style-guide/index.html
##[group]Documenting stage2 releases (aarch64-unknown-linux-gnu)
##[endgroup]
##[group]Documenting stage2 releases (aarch64-unknown-linux-gnu -> i686-unknown-haiku)
##[endgroup]
##[group]Building stage1 tool rust-installer (aarch64-unknown-linux-gnu)
##[endgroup]
Dist rust-docs-1.90.0-dev-aarch64-unknown-linux-gnu
Dist rust-docs-1.90.0-dev-i686-unknown-haiku
##[group]Documenting stage2 library {alloc, compiler_builtins, core, panic_abort, panic_unwind, proc_macro, rustc-std-workspace-core, std, std_detect, sysroot, test, unwind} in JSON format (aarch64-unknown-linux-gnu)
##[endgroup]
Dist rust-docs-json-1.90.0-dev-aarch64-unknown-linux-gnu
##[group]Documenting stage2 library {alloc, compiler_builtins, core, panic_abort, panic_unwind, proc_macro, rustc-std-workspace-core, std, std_detect, sysroot, test, unwind} in JSON format (aarch64-unknown-linux-gnu -> i686-unknown-haiku)
##[endgroup]
Dist rust-docs-json-1.90.0-dev-i686-unknown-haiku
Vendoring sources to "/checkout"
##[group]Building stage1 tool generate-copyright (aarch64-unknown-linux-gnu)
##[endgroup]
Dist rustc-1.90.0-dev-aarch64-unknown-linux-gnu
##[group]Building LLVM for i686-unknown-haiku
##[endgroup]
##[group]Building compiler artifacts (stage1:aarch64-unknown-linux-gnu -> stage2:i686-unknown-haiku)
##[endgroup]
Creating a sysroot for stage2 compiler (i686-unknown-haiku) (use `rustup toolchain link 'name' build/i686-unknown-haiku/stage2`)
##[group]Building tool rustdoc_tool_binary (stage2:aarch64-unknown-linux-gnu -> stage3:i686-unknown-haiku)
##[endgroup]
Dist rustc-1.90.0-dev-i686-unknown-haiku
Dist rust-std-1.90.0-dev-aarch64-unknown-linux-gnu
Dist rust-std-1.90.0-dev-i686-unknown-haiku
Dist rustc-dev-1.90.0-dev-aarch64-unknown-linux-gnu
Dist rustc-dev-1.90.0-dev-i686-unknown-haiku
Dist rust-src-1.90.0-dev
Dist rust-dev-1.90.0-dev-aarch64-unknown-linux-gnu
Dist rust-dev-1.90.0-dev-i686-unknown-haiku
Dist rustc-1.90.0-dev-src
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ Snapshot Summary ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
Snapshot: dist_with_targets_and_hosts
Source: src/bootstrap/src/core/builder/tests.rs:1190
────────────────────────────────────────────────────────────────────────────────
Expression: ctx.config("dist").hosts(&[&host_target(),
TEST_TRIPLE_1]).targets(&[&host_target(), TEST_TRIPLE_1]).render_steps()
────────────────────────────────────────────────────────────────────────────────
-old snapshot
+new results
────────────┬───────────────────────────────────────────────────────────────────
    3     3 │ [build] rustc 0 <host> -> rustc 1 <host>
    4     4 │ [build] rustc 1 <host> -> std 1 <host>
    5     5 │ [build] rustc 1 <host> -> rustc 2 <host>
    6     6 │ [build] rustdoc 1 <host>
    7       │-[doc] std 2 <host> crates=[alloc,compiler_builtins,core,panic_abort,panic_unwind,proc_macro,std,std_detect,sysroot,test,unwind]
    8       │-[doc] std 2 <target1> crates=[alloc,compiler_builtins,core,panic_abort,panic_unwind,proc_macro,std,std_detect,sysroot,test,unwind]
          7 │+[doc] std 2 <host> crates=[alloc,compiler_builtins,core,panic_abort,panic_unwind,proc_macro,rustc-std-workspace-core,std,std_detect,sysroot,test,unwind]
          8 │+[doc] std 2 <target1> crates=[alloc,compiler_builtins,core,panic_abort,panic_unwind,proc_macro,rustc-std-workspace-core,std,std_detect,sysroot,test,unwind]
    9     9 │ [build] rustc 2 <host> -> std 2 <host>
   10    10 │ [build] rustc 0 <host> -> LintDocs 1 <host>
   11    11 │ [build] rustc 1 <host> -> std 1 <target1>
   12    12 │ [build] rustc 2 <host> -> std 2 <target1>
────────────┴───────────────────────────────────────────────────────────────────
Stopped on the first failure. Run `cargo insta test` to run all snapshots.

thread 'core::builder::tests::snapshot::dist_with_targets_and_hosts' panicked at /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/insta-1.43.1/src/runtime.rs:679:13:
snapshot assertion for 'dist_with_targets_and_hosts' failed in line 1190
---
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ Snapshot Summary ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
Snapshot: doc_library
Source: src/bootstrap/src/core/builder/tests.rs:1616
────────────────────────────────────────────────────────────────────────────────
Expression: ctx.config("doc").path("library").render_steps()
────────────────────────────────────────────────────────────────────────────────
-old snapshot
+new results
────────────┬───────────────────────────────────────────────────────────────────
    0     0 │ [build] llvm <host>
    1     1 │ [build] rustc 0 <host> -> rustc 1 <host>
    2     2 │ [build] rustdoc 0 <host>
    3       │-[doc] std 1 <host> crates=[alloc,compiler_builtins,core,panic_abort,panic_unwind,proc_macro,std,std_detect,sysroot,test,unwind]
          3 │+[doc] std 1 <host> crates=[alloc,compiler_builtins,core,panic_abort,panic_unwind,proc_macro,rustc-std-workspace-core,std,std_detect,sysroot,test,unwind]
────────────┴───────────────────────────────────────────────────────────────────
Stopped on the first failure. Run `cargo insta test` to run all snapshots.

thread 'core::builder::tests::snapshot::doc_library' panicked at /cargo/registry/src/index.crates.io-1949cf8c6b5b557f/insta-1.43.1/src/runtime.rs:679:13:
snapshot assertion for 'doc_library' failed in line 1616


failures:
    core::builder::tests::snapshot::alias_and_path_for_library
    core::builder::tests::snapshot::dist_all_cross
```

</details>

bjorn3 commented:
@bors r+

bors commented:
:pushpin: Commit d7654b920f71d0cf50f5089ec3a48734370e08af has been approved by `bjorn3`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=bjorn3 d7654b920f71d0cf50f5089ec3a48734370e08af -->
<!-- homu: {"type":"Approved","sha":"d7654b920f71d0cf50f5089ec3a48734370e08af","approver":"bjorn3","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

Noratrieb commented:
@bors r-
failed in rollup because UEFI does funny things with compiler-builtins that I imagine one is not supposed to do: https://github.com/rust-lang/rust/pull/144709#issuecomment-3138582433

tgross35 committed:
uefi: Use slice equality rather than `memcmp`

`compiler_builtins` shouldn't be called directly. Change the `PartialEq`
implementation for `DevicePathNode` to use slice equality instead, which
will call `memcmp`/`bcmp` via the intrinsic.

Noratrieb reviewed (approved):
on the uefi change

bors commented:
:pushpin: Commit 42bf044ce83a060bfb16694ff49dc92784abf0a0 has been approved by `bjorn3,Noratrieb`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=bjorn3,Noratrieb 42bf044ce83a060bfb16694ff49dc92784abf0a0 -->
<!-- homu: {"type":"Approved","sha":"42bf044ce83a060bfb16694ff49dc92784abf0a0","approver":"bjorn3,Noratrieb","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->
